src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-sync-pipe-modal/rgw-multisite-sync-pipe-modal.component.ts

```diff
@@ -119,20 +124,47 @@ export class RgwMultisiteSyncPipeModalComponent implements OnInit {
       this.pipeForm.setErrors({ cdSubmitButton: true });
       return;
     }
-    this.rgwMultisiteService.createEditSyncPipe(this.pipeForm.getRawValue()).subscribe(
-      () => {
-        const action = this.editing ? 'Modified' : 'Created';
-        this.notificationService.show(
-          NotificationType.success,
-          $localize`${action} Sync Pipe '${this.pipeForm.getValue('pipe_id')}'`
-        );
-        this.activeModal.close('success');
-      },
-      () => {
-        // Reset the 'Submit' button.
-        this.pipeForm.setErrors({ cdSubmitButton: true });
-        this.activeModal.dismiss();
-      }
-    );
+
+    const sourceZones: any = { added: [], removed: [] };
+    const destZones: any = { added: [], removed: [] };
+    if (this.editing) {
+      destZones.removed = this.pipeSelectedRow.dest.zones.filter(
+        (zone: string) => !this.destZones.data.selected.includes(zone)
+      );
+      destZones.added = this.destZones.data.selected.filter(
+        (zone: string) => !this.pipeSelectedRow.dest.zones.includes(zone)
+      );
+      sourceZones.removed = this.pipeSelectedRow.source.zones.filter(
+        (zone: string) => !this.sourceZones.data.selected.includes(zone)
+      );
+      sourceZones.added = this.sourceZones.data.selected.filter(
+        (zone: string) => !this.pipeSelectedRow.source.zones.includes(zone)
+      );
+    }
+    sourceZones.added =
+      sourceZones.added.length > 0 ? sourceZones.added : this.sourceZones.data.selected;
+    destZones.added = destZones.added.length > 0 ? destZones.added : this.destZones.data.selected;
+
+    this.rgwMultisiteService
+      .createEditSyncPipe({
+        ...this.pipeForm.getRawValue(),
+        source_zones: sourceZones,
+        destination_zones: destZones
+      })
+      .subscribe(
+        () => {
+          const action = this.editing ? 'Modified' : 'Created';
```

Comment:
is it locallzed/i18n ? I think we have a set of action constants which you may use.

Reply:
yes it is

Reply:
What about useing "TableActions"  defined constants here ?

Reply:
IMO 'edit' is a action and not 'modified', also create is an action and it doesn't makes sense to add create'd' in actions list.